cpp/lib/src/Downloader.cc

```diff
@@ -711,6 +711,21 @@ void Downloader::setUserAgent(const std::string &user_agent) {
 }
 
 
+bool Downloader::getHttpRedirectedUrl(std::string * const redirected_url) {
+    if (params_.follow_redirects_)
+        throw std::runtime_error(__FUNCTION__ + std::string(" Cannot determine redirect url if automatic redirection is enabled"));
+    //We only address 3XX HTTP codes
+    if (getResponseCode() / 100 != 3)
+        return false;
+    char *location(0);
+    curl_error_code_ = ::curl_easy_getinfo(getEasyHandle(), CURLINFO_REDIRECT_URL, &location);
```

Comment:
Why don't you pass in redirected_url instead of &location?

Reply:
Apparently curl_easy_getinfo allocates and manages memory internally

Reply:
Accepted, but, don't we have to release this memory?

Reply:
According to https://curl.haxx.se/libcurl/c/curl_easy_getinfo.html obviously not.

Reply:
i.e. is seemingly done by curl_easy_cleanup in the destructor.